panel/tests/ui/widgets/test_player.py

```diff
@@ -53,3 +53,45 @@ def test_name_and_show_value(page):
 
     name = page.locator('.pn-player-title:has-text("test")')
     expect(name).to_have_count(1)
+def test_player_visible_buttons(page):
+    player = Player(visible_buttons=["play", "pause"])
+    serve_component(page, player)
+
+    assert page.is_visible(".play")
+    assert page.is_visible(".pause")
+    assert not page.is_visible(".reverse")
+    assert not page.is_visible(".first")
+    assert not page.is_visible(".previous")
+    assert not page.is_visible(".next")
+    assert not page.is_visible(".last")
+    assert not page.is_visible(".slower")
+    assert not page.is_visible(".faster")
+
+    player.visible_buttons = ["first"]
+    wait_until(lambda: page.is_visible(".first"))
```

Comment:
`wait_until` is meant for waiting for some condition in Python to be met. For checking things on the frontend you should use the `expect` API:

```suggestion
    expect(page.locator(".first")).to_be_visible()
```